### PR TITLE
Fixes from VPP datapath bring-up

### PR DESCRIPTION
**Pull request by shastrinator**
Abstract direct accesses to integration bridge
Fix VPP package paths since package was relocated
Fix return value in VPP api
Fix unavailable constant in Ocata

### CODE REVIEW

File: opflexagent/vif_plug_vpp/linux_net.py [hunk 1]
```diff
@@ -42,7 +41,7 @@ def _create_vpp_vif(dev, iface_id, mac, mtu,
                         1, mac_address, iface_id)
     LOG.debug("sw_if_index:{}".format(sw_if_index))
     if mtu:
-        vapi.set_interface_mtu(sw_if_index, mtu)
+        vapi.set_interface_mtu(sw_if_index, [mtu, 0, 0, 0])
```
Comment (kentwu1) [inline opflexagent/vif_plug_vpp/linux_net.py]: not sure what the extra 3 0's mean here.....

File: opflexagent/vpplib/VPPApi.py [hunk 2]
```diff
@@ -49,8 +49,7 @@ def __init__(self, client_name, LOG):
         self.connected = False
 
     def __enter__(self):
-        self.vppp = VppPapiProvider(self.client_name, self.LOG,
-            None, self.read_timeout)
+        self.vppp = VppPapiProvider(self.client_name, None, self.read_timeout)
```
Comment (kentwu1) [inline opflexagent/vpplib/VPPApi.py]: Wondering how this error can get away with it before.....Don't we have a UT in place to exercise this code path then?

File: opflexagent/vpp_plug_firewall.py [hunk 3]
```diff
@@ -28,8 +28,7 @@ def vpp_nova_to_osvif_vif_vhostuser(vif):
         except NotImplementedError as e:
             if vif['details'].get("vhostuser_vpp_plug", False):
                 profile = objects.vif.VIFPortProfileOpenVSwitch(
-                    interface_id=vif.get('ovs_interfaceid') or vif['id'],
-                    datapath_type=model.VIF_DETAILS_OVS_DATAPATH_NETDEV)
+                    interface_id=vif.get('ovs_interfaceid') or vif['id'])
```
Comment (kentwu1) [inline opflexagent/vpp_plug_firewall.py]: not that familiar with the code but why we don't need to pass in the datapath_type parameter now?

File: opflexagent/vpplib/VPPApi.py (hunk 2 above)
Comment (shastrinator) [reply opflexagent/vpplib/VPPApi.py]: I have used mock APIs at this level in the UT intentionally. We were not sure about whether we would release this library or the VPP team. I had all these changes locally and missed committing  the last time around, which was why I retested this path.

File: opflexagent/vif_plug_vpp/linux_net.py (hunk 1 above)
Comment (shastrinator) [reply opflexagent/vif_plug_vpp/linux_net.py]: 4 values are for [all packets, L3 packets, IPv4 packets, MPLS packets], first value applies to all unless overriden.

File: opflexagent/vpp_plug_firewall.py (hunk 3 above)
Comment (shastrinator) [reply opflexagent/vpp_plug_firewall.py]: This function derives from nova/nova/network/os_vif_util.py _nova_to_osvif_vif_vhostuser. This argument was added post ocata. I had to make this change for testing in ocata. For the master branch, I can have this argument still, I will add it back. Thanks.

File: opflexagent/snat_iptables_manager.py [hunk 4]
```diff
@@ -27,13 +27,13 @@
...
-    def __init__(self, int_br):
-        self.int_br = int_br
+    def __init__(self, bridge_manager):
+        self.bridge_manager = bridge_manager
         self.snat_conn_track_handler = (
-            as_metadata_manager.SnatConnTrackHandler())
+            as_metadata_manager.SnatConnTrackHandler(bridge_manager))
 
     def _cleanup(self, if_name, ns_name):
-        self.int_br.delete_port(if_name)
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply opflexagent/snat_iptables_manager.py]: Sure, I was misled by the same name. I will change them to use fabric_br. No UT failed though. Thanks for catching this.

File: opflexagent/utils/bridge_managers/ovs_manager.py [hunk 5]
```diff
@@ -170,6 +170,18 @@ def get_vif_port_by_id(self, id):
     def handle_removed_eps(self, em, removed_eps):
         return removed_eps
 
+    def plug_metadata_port(self, dst_shell, port):
+        dst_shell("ovs-vsctl add-port %s %s" % (self.fabric_br.br_name, port))
+
+    def delete_port(self, port):
+        self.int_br.delete_port(port)
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply opflexagent/utils/bridge_managers/ovs_manager.py]: Sure

**Comment by kentwu1** [comment]
It looks ok to me now but have you run some sanity test on a real opflex fab to make sure its working as expected? Thx.

**Comment by tbachman** [comment]
The pep8 tox job runs the DB migration, and that worked.

**Pull request merged**